### PR TITLE
CLN: Migrate resizeable vector types and functions to dedicated files

### PR DESCRIPTION
**Pull request by brendan-ward**
This moves resizeable vector-related typedefs and functions to `vector.h` / `vector.c` for more generalized use elsewhere.  These currently were defined within strtree, which was their only use so far.

However, for #244 and likely other functions over the long term, we will need more general access to these so that we can build up vectors at runtime.

Types and functions renamed slightly for better clarity (I hope!).

### CODE REVIEW

**Comment by brendan-ward** [comment]
Moving to draft: found a few changes needed to better align with standard naming conventions (add `_t` suffix to types).

In order to use in Cython, we also need to provide concrete types (can't pass type as a parameter to functions like `kv_push` from Cython).  I think this is solved by providing a few additional macros for each vector type we use here, which gives us an opportunity to make the functions we call named a little bit more clearly (`kv_A` is pretty non-obvious).

File: src/vector.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#ifndef _VECTOR_H
+#define _VECTOR_H
+
+#include <Python.h>
+#include <numpy/ndarraytypes.h>
+
+#include "geos.h"
```
Comment (caspervdw) [inline src/vector.h:7]: Only put includes in a header if the header itself needs them. I think you should drop ` geos.h` and `kvec.h` here, only `pygeom.h` is needed for the `GeometryObject` .

File: src/vector.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#include <structmember.h>
+
+#define NO_IMPORT_ARRAY
+#define PY_ARRAY_UNIQUE_SYMBOL pygeos_ARRAY_API
+
+#include <numpy/arrayobject.h>
+#include <numpy/ndarraytypes.h>
+#include <numpy/npy_3kcompat.h>
+
+#include "geos.h"
```
Comment (caspervdw) [inline src/vector.c:14]: There are some includes here that are not needed: I think only `kvec.h` is needed here.

**Review by caspervdw — APPROVED** [review]
Some minor remarks about includes, for the rest, feel free to merge this

**Comment by caspervdw** [comment]
Thanks @brendan-ward

**Pull request merged**